gravity-forms/gw-validate-current-password.php

```diff
@@ -0,0 +1,16 @@
+<?php
+/**
+ * Gravity Wiz // Gravity Forms // Validate Current Password
+ * http://gravitywiz.com/
+ */
+// update "123" to your form ID, update "4" to your Password field ID
+add_filter( 'gform_field_validation_123_4', function( $result, $value ) {
+	$user = wp_get_current_user();
+	if ($user){
+		if ( ! wp_check_password( $value, $user->data->user_pass, $user->ID ) ) {
+			$result['is_valid'] = false;
+			$result['message'] = 'Invalid current password. Please try again.';
+		}
+	}
```

Comment:
Let's simplify this to a single line check.
```
if ( ! $user || ! wp_check_password( $value, $user->data->user_pass, $user->ID ) ) { 
```

Reply:
I just updated it to a single line check.

Reply:
Haha, don't hate me but I gave you some bad logic. 😆

This should only happen if the user is logged in and the password is bad.

```
if ( $user && ! wp_check_password( $value, $user->data->user_pass, $user->ID ) ) { 
```

Reply:
Yes, you're right. That fixes it.